src/rebar3_hex_build.erl

```diff
@@ -1,3 +1,88 @@
+% @doc `rebar3 hex build' 
+%
+%% Builds a new local version of your package.
+%%
+%% By default this provider will build both a package tarball and docs tarball. 
+%%  
+%% The package and docs .tar files are created in the current directory, but is not pushed to the repository. An app 
+%% named foo at version 1.2.3 will be built as foo-1.2.3.tar. Likewise the docs .tar would be built as 
+%% foo-1.2.4-docs.tar. 
+%%
+%% ```shell
+%% $ rebar3 hex build 
+%% '''
+%% 
+%% You may also build only a package or docs tarball utilizing the same available command line options. 
+%%
+%% ``` shell
+%% $ rebar3 hex build package 
+%% '''
+%%
+%% ```
+%% $ rebar3 hex build docs 
+%% '''
+%% 
+%% <h2>Configuration</h2> 
+%% Packages are configured via `src/<myapp>.app.src'  attributes.
+%% 
+%% == Required configuration == 
+%% 
+%% <ul>
+%%  <li> `application' - application name. This is required per Erlang/OTP thus it should always be present anyway. 
+%   <li> `vsn' -  must be a valid [semantic version](http://semver.org/) identifier.
+%%  <li>`licenses' - A list of licenses the project is licensed under. This attribute is required. A valid
+%%  [spdx](https://spdx.org/licenses/) is expected.</li>
+%% </ul>
+
+%% 
+%% == Optional configuration == 
+%% In addition, the following meta attributes are supported and highly recommended : 
+%%
+%% <ul>
+%%  <li> `description' - a brief description about your application.
+%%  <li>`pkg_name' - The name of the package in case you want to publish the package with a different name than the
+%%  application name.</li>
+%%  <li>`links' - A map where the key is a link name and the value is the link URL. Optional but highly recommended.
+%%  <li> `files' - A list of files and directories to include in the package. Defaults to standard project directories, 
+%%        so you usually don't need to set this property.</li>
+%%  <li> `include_files' - A list of paths containing files you wish to include in a release. </li>
```

Comment:
I feel like these should be called `include_paths` and `exclude_paths` and we could also perhaps support `exclude_files`. They both work on paths right now. Thoughts?

Reply:
the `_paths` variants are clearer, but it's not a super significant difference either way, especially if this ends up using glob or wildcard patterns (though that's `exclude_patterns`, which has also been renamed). The "confusion" is already common (e.g. `filelib` supports `is_file|dir|regular` already). 

I'd tend to still lower the updating friction for the plugin by keeping this compatible with the stuff that was already in the .app.src file though, to make upgrading the plugin as easy as possible. The 7.0 update will likely have people update their licenses, so there should at least be an obvious way to know that you're done fixing what broke rather than suddenly having older options not work (if any, I mostly recall just using `files`).

Either way I'm fine with the direction we're going.